src/network/networkd-ndisc.c

```diff
@@ -199,6 +206,13 @@ static int ndisc_request_route(Route *in, Link *link, sd_ndisc_router *rt) {
         if (route->mtu == 0)
                 route->mtu = mtu;
 
+        if (link->network->ipv6_hop_limit < 0 && hop_limit > 0) {
```

Comment:
Avoid a configuration option when an RFC specifies the feature as a "SHOULD" requirement.